tests/test_MocLogin.py

```diff
@@ -0,0 +1,107 @@
+from django.test import TestCase, LiveServerTestCase
+from core.models import AtmosphereUser as User
+from core.models import Provider
+from core.models import PlatformType
+from core.models import ProviderType
+
+from threepio import logger
+
+from atmosphere import settings
+from atmosphere.settings.local import AUTHENTICATION as auth_settings
+from atmosphere.settings.local import TEST as test_settings
+
+import requests
+
+
+# This is assuming a clean database
+class MocLogin(LiveServerTestCase):
+    username = 'MockeyMock'
+    password = 'MockeyMock'
+
+    def _setup_provider(self):
+        # Need to set up the default provider
+        kvm = PlatformType.objects.get_or_create(name='KVM')[0]
+        openstack_type = ProviderType.objects.get_or_create(name='OpenStack')[0]
+        MOC = Provider.objects.get_or_create(
+            location="MOC-Test",
+            description="MOC-Test",
+            virtualization=kvm,
+            type=openstack_type,
+            public=True)[0]
+        MOC.cloud_config = \
+            '{"user": {"domain": "default", "secret": "012345678901234567890123456789012",'\
+            ' "user_role_name": "_member_", "admin_role_name": "admin"},'\
+            ' "deploy": {"hostname_format": "%(one)s.%(two)s.%(three)s.%(four)s"},'\
+            ' "network": {"topology": "External Network Topology",'\
+            '     "dns_nameservers": ["8.8.8.8", "8.8.4.4"],'\
+            '     "default_security_rules": [["ICMP", -1, -1], ["UDP", 20, 20], ["TCP", 20, 21], ["TCP", 22, 23],'\
```

Comment:
OK, I have clipped them to a subset.  Should have a few so that we accurately represent what the cloud_config contains.